### PR TITLE
License ns updates

### PR DESCRIPTION
**Pull request by tjasmith**
Updates on the license namespace project

- Promote license namespace functionality
- Add license list url and github repo URL
- Add spdx file url to submitted namespace issue

### CODE REVIEW

**Comment by tjasmith** [comment]
@techytushar please could you have a look?

File: src/src/settings.py [hunk 1]
```diff
@@ -20,12 +20,15 @@
...
+LICENSE_REPO_NAME = "license-list-XML"
+LICENSE_TEST_REPO_NAME = "TEST-LicenseList-XML"
+DEV_REPO_URL = 'https://api.github.com/repos/spdx/{0}'.format(LICENSE_TEST_REPO_NAME)
+PROD_REPO_URL = 'https://api.github.com/repos/spdx/{0}'.format(LICENSE_REPO_NAME)
 REPO_URL = DEV_REPO_URL
 
-NAMESPACE_DEV_REPO_URL = 'https://api.github.com/repos/tjasmith/license-namespace-list/issues'
-NAMESPACE_PROD_REPO_URL = 'https://api.github.com/repos/tjasmith/license-namespace-list/issues'
+NAMESPACE_REPO_NAME = "license-namespace-list"
+NAMESPACE_DEV_REPO_URL = 'https://api.github.com/repos/tjasmith/{0}'.format(NAMESPACE_REPO_NAME)
```
Comment (goneall) [inline src/src/settings.py]: Should I create a test repo in SPDX for this?

Comment (tjasmith) [reply src/src/settings.py]: Yes,

Comment (goneall) [reply src/src/settings.py]: I created to repos and added tjasmith with write access:

- https://github.com/spdx/license-namespace - for production
- https://github.com/spdx/license-namespace-test - for test

File: src/app/views.py [hunk 2]
```diff
@@ -331,6 +343,63 @@ def licenseInformation(request, licenseId):
...
+    licenseInformation['licenseAuthorName'] = licenseNamespaceRequest.licenseAuthorName
+    licenseInformation['archive'] = licenseNamespaceRequest.archive
+
+    licenseInformation['notes'] = licenseNamespaceRequest.notes
+    licenseInformation['namespace'] = licenseNamespaceRequest.namespace
+    licenseInformation['url'] = licenseNamespaceRequest.url
+    licenseInformation['description'] = licenseNamespaceRequest.description
+    licenseInformation['publiclyShared'] = licenseNamespaceRequest.publiclyShared
+    xmlString = licenseNamespaceRequest.xml
+    data = parseXmlString(xmlString)
```
Comment (techytushar) [inline src/app/views.py]: This function is defined in `utils`, I guess this is the reason the checks are failing

File: src/app/views.py [hunk 3]
```diff
@@ -1190,6 +1286,75 @@ def archiveRequests(request, license_id=None):
...
+    return render(request,
+        'app/archive_namespace_requests.html',context_dict
+        )
+
+
+def promoteNamespaceRequests(request, license_id=None):
+    """ View for promote namespace license requests
+    returns promote_namespace_requests.html template
+    """
+    from django.forms import model_to_dict
```
Comment (techytushar) [inline src/app/views.py]: its better to do the import at the top

File: src/app/views.py [hunk 4]
```diff
@@ -1274,6 +1457,61 @@ def pull_request(request):
...
+            ajaxdict = {}
+            try:
+                if request.user.is_authenticated():
+                    user = request.user
+                try:
+                    """ Getting user info and calling the makePullRequest function """
+                    github_login = user.social_auth.get(provider='github')
+                    token = github_login.extra_data["access_token"]
+                    username = github_login.extra_data["login"]
+                    response = utils.makeNsPullRequest(username, token, request.POST["branchName"], request.POST["updateUpstream"], request.POST["fileName"], request.POST["commitMessage"], request.POST["prTitle"], request.POST["prBody"], request.POST["xmlText"])
```
Comment (techytushar) [inline src/app/views.py]: better to modify the existing function.

File: src/app/utils.py [hunk 5]
```diff
@@ -191,6 +193,151 @@ def makePullRequest(username, token, branchName, updateUpstream, fileName, commi
         "pr_url": data["html_url"],
     }
 
+def makeNsPullRequest(username, token, branchName, updateUpstream, fileName, commitMessage, prTitle, prBody, xmlText):
```
[the remark this answers is not in the document]
Comment (techytushar) [reply src/app/utils.py]: @tjasmith update this function also so that we can merge this PR

**Comment by tjasmith** [comment]
@techytushar Please have you had a look at the update?

**Comment by techytushar** [comment]
Yes, I have looked at the updates they all seem fine to me except the `makeNsPullRequest` function. Can you merge this function with the already existing `makePullRequest` function since they both have mostly the same code. Do this change and then we'll merge this. 
Also for the travis builds there seem to be a problem with `xvfb` which right now I can't figure out how to solve but don't worry about that, I am working on it.

**Comment by tjasmith** [comment]
@techytushar @rtgdk I will write them. But please could you merge this PR so I can proceed? This PR has been here for so long.

**Comment by rtgdk** [comment]
@tjasmith LGTM. Merged.

**Comment by tjasmith** [comment]
@rtgdk Thanks

**Pull request merged**